From 9ccef963bc825075db6d6b6458bdfde1ebcab6d1 Mon Sep 17 00:00:00 2001
From: Ryan Lane <laner@users.mediawiki.org>
Date: Thu, 15 Mar 2012 00:04:49 +0000
Subject: [PATCH] Reverting AuthAttribute back to former implementation. I
 really need to read my own documentation before I make
 stupid changes like this.

---
 LdapAuthentication.php |   11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

--- a/dist/mediawiki-extensions-ldapauth/usr/share/mediawiki-extensions/ldapauth/LdapAuthentication.php
+++ b/dist/mediawiki-extensions-ldapauth/usr/share/mediawiki-extensions/ldapauth/LdapAuthentication.php
@@ -677,7 +677,11 @@ class LdapAuthenticationPlugin extends A
 			$aa = $this->getConf( 'AuthAttribute' ); 
 			if ( $aa ) {
 				$this->printDebug( "Checking for auth attributes: $aa", NONSENSITIVE );
-				if ( !isset( $this->userInfo ) || !isset( $this->userInfo[0][$aa] ) ) {
+				$filter = "(" . $aa . ")";
+				$attributes = array( "dn" );
+				$entry = LdapAuthenticationPlugin::ldap_read( $this->ldapconn, $this->userdn, $filter, $attributes );
+				$info = LdapAuthenticationPlugin::ldap_get_entries( $this->ldapconn, $entry );
+				if ( $info["count"] < 1 ) {
 					$this->printDebug( "Failed auth attribute check", NONSENSITIVE );
 					LdapAuthenticationPlugin::ldap_unbind( $this->ldapconn );
 					$this->markAuthFailed();
@@ -998,11 +1002,6 @@ class LdapAuthenticationPlugin extends A
 				return false;
 			}
 
-			$aa = $this->getConf( 'AuthAttribute' );
-			if ( $aa ) {
-				$values[$aa] = "true";
-			}
-
 			$this->printDebug( "Adding user", NONSENSITIVE );
 			if ( LdapAuthenticationPlugin::ldap_add( $this->ldapconn, $this->userdn, $values ) ) {
 				$this->printDebug( "Successfully added user", NONSENSITIVE );
